shared/actions/teams.js

```diff
@@ -958,102 +947,118 @@ const _setMemberPublicity = function*(action: TeamsGen.SetMemberPublicityPayload
+                joinAs: RPCTypes.teamsTeamRole[settings.openTeamRole],
+                open: settings.openTeam,
+              },
+            },
+            waitingKey
+          )
+            .then(payload => ({payload, type: 'ok'}))
+            .catch(payload => ({payload, type: 'error'}))
         })
-          .then(payload => ({type: 'ok', payload}))
-          .catch(payload => ({type: 'error', payload}))
-      })
-    )
-  }
-  if (publicityAnyMember !== settings.publicityAnyMember) {
-    calls.push(
-      Saga.call(function*() {
-        RPCTypes.teamsSetTeamShowcaseRpcPromise({
-          anyMemberShowcase: settings.publicityAnyMember,
-          name: teamname,
+      )
+    }
+    if (ignoreAccessRequests !== settings.ignoreAccessRequests) {
+      calls.push(
+        Saga.call(function*() {
+          return RPCTypes.teamsSetTarsDisabledRpcPromise(
+            {
+              disabled: settings.ignoreAccessRequests,
+              name: teamname,
+            },
+            waitingKey
+          )
+            .then(payload => ({payload, type: 'ok'}))
+            .catch(payload => ({payload, type: 'error'}))
         })
-          .then(payload => ({type: 'ok', payload}))
-          .catch(payload => ({type: 'error', payload}))
-      })
-    )
-  }
-  if (publicityMember !== settings.publicityMember) {
-    calls.push(
-      Saga.call(function*() {
-        RPCTypes.teamsSetTeamMemberShowcaseRpcPromise({
-          isShowcased: settings.publicityMember,
-          name: teamname,
+      )
+    }
+    if (publicityAnyMember !== settings.publicityAnyMember) {
+      calls.push(
+        Saga.call(function*() {
+          return RPCTypes.teamsSetTeamShowcaseRpcPromise(
+            {
+              anyMemberShowcase: settings.publicityAnyMember,
+              name: teamname,
+            },
+            waitingKey
+          )
+            .then(payload => ({payload, type: 'ok'}))
+            .catch(payload => ({payload, type: 'error'}))
         })
-          .then(payload => ({type: 'ok', payload}))
-          .catch(payload => ({type: 'error', payload}))
-      })
-    )
-  }
-  if (publicityTeam !== settings.publicityTeam) {
-    calls.push(
-      Saga.call(function*() {
-        RPCTypes.teamsSetTeamShowcaseRpcPromise({
-          isShowcased: settings.publicityTeam,
-          name: teamname,
+      )
+    }
+    if (publicityMember !== settings.publicityMember) {
+      calls.push(
+        Saga.call(function*() {
+          return RPCTypes.teamsSetTeamMemberShowcaseRpcPromise(
+            {
+              isShowcased: settings.publicityMember,
+              name: teamname,
+            },
+            waitingKey
+          )
+            .then(payload => ({payload, type: 'ok'}))
+            .catch(payload => ({payload, type: 'error'}))
         })
-          .then(payload => ({type: 'ok', payload}))
-          .catch(payload => ({type: 'error', payload}))
-      })
-    )
-  }
-  return Saga.all([
```

Comment:
doesn't actually make sense that this was .all. this means the increment and calls and the last call were all racing for some reason